src/ui/public/vis/agg_config.js

```diff
@@ -148,13 +148,11 @@ export default function AggConfigFactory(Private, fieldTypeFilter) {
    * @return {object} the new params object
    */
   AggConfig.prototype.resetParams = function () {
-    let fieldParam = this.type && this.type.params.byName.field;
     let field;
+    const fieldOptions = this.getFieldOptions();
 
-    if (fieldParam) {
-      let prevField = this.params.field;
-      let fieldOpts = fieldTypeFilter(this.vis.indexPattern.fields, fieldParam.filterFieldTypes);
-      field = _.contains(fieldOpts, prevField) ? prevField : null;
+    if (fieldOptions) {
+      field = fieldOptions.byName[this.params.field.name] || null;
```

Comment:
Getting an undefined error on this line when I select an agg from the dropdown:

![screen shot 2016-10-18 at 11 31 25 am](https://cloud.githubusercontent.com/assets/6239176/19484797/d066ae80-9526-11e6-8476-426e7bf24806.png)